test/core/rtsmith_toml_test.cpp

```diff
@@ -0,0 +1,374 @@
+    // Check if the `programInfo` object is generated successfully.
+    ASSERT_TRUE(programInfo != nullptr);
+
+    // Parse the fuzzer configurations from the TOML file.
+    toml::parse_result tomlConfig;
+    try {
+        tomlConfig = toml::parse_file(configFilePath);
+    } catch (const toml::parse_error &e) {
+        error("P4RuntimeSmith: Failed to parse fuzzer configuration file: %1%", e.what());
+    }
+    int maxEntryGenCntConfig;
+    int maxAttemptsConfig;
+    int maxTablesConfig;
+    std::vector<std::string> tablesToSkipConfig;
+    uint64_t thresholdForDeletionConfig;
+    size_t maxUpdateCountConfig;
+    uint64_t maxUpdateTimeInMicrosecondsConfig;
+    uint64_t minUpdateTimeInMicrosecondsConfig;
+    auto maxEntryGenCntNode = tomlConfig["maxEntryGenCnt"];
+    // Check if the node exists.
+    if (!!maxEntryGenCntNode) {
+        // Check if the node represents an integer, meaning whether the configuration provided is
+        // valid.
+        if (maxEntryGenCntNode.type() == toml::node_type::integer) {
+            maxEntryGenCntConfig = maxEntryGenCntNode.value<int>().value();
+        } else {
+            error("P4RuntimeSmith: The maximum number of entries to generate must be an integer.");
+        }
+    } else {
+        // If the node does not exist, throw an error that indicates the configuration must be
+        // provided.
+        error("P4RuntimeSmith: The maximum number of entries to generate must be provided.");
+    }
+    auto maxAttemptsNode = tomlConfig["maxAttempts"];
+    if (!!maxAttemptsNode) {
+        if (maxAttemptsNode.type() == toml::node_type::integer) {
+            maxAttemptsConfig = maxAttemptsNode.value<int>().value();
+        } else {
+            error("P4RuntimeSmith: The maximum number of attempts must be an integer.");
+        }
+    } else {
+        error("P4RuntimeSmith: The maximum number of attempts must be provided.");
+    }
+    auto maxTablesNode = tomlConfig["maxTables"];
+    if (!!maxTablesNode) {
+        if (maxTablesNode.type() == toml::node_type::integer) {
+            maxTablesConfig = maxTablesNode.value<int>().value();
+        } else {
+            error("P4RuntimeSmith: The maximum number of tables must be an integer.");
+        }
+    } else {
+        error("P4RuntimeSmith: The maximum number of tables must be provided.");
+    }
+    auto tablesToSkipNode = tomlConfig["tablesToSkip"];
+    if (!!tablesToSkipNode) {
+        if (tablesToSkipNode.type() == toml::node_type::array) {
+            const auto *expectedStringRepresentations = tablesToSkipNode.as_array();
+            for (const auto &expectedStringRepresentation : *expectedStringRepresentations) {
+                if (const auto *str = expectedStringRepresentation.as_string()) {
+                    tablesToSkipConfig.push_back(str->get());
+                } else {
+                    error("P4RuntimeSmith: The tables to skip must be strings.");
```

Comment:
Do not use errors in gtest, instead uses assertions: http://google.github.io/googletest/reference/assertions.html